packages/lab/src/date-picker/useDatePicker.ts

```diff
@@ -227,19 +266,27 @@ export function useDatePicker<
       details: DateInputSingleDetails,
     ) => {
       setSelectedDate(date);
+      checkAndAddError(date, isDayDisabled, "disabled", details);
+      checkAndAddError(date, isDayUnselectable, "unselectable", details);
```

Comment:
when will the user select unselectable date?

Reply:
they can edit the input and enter an unselectable date, which with this code, creates an error that the formfield can use.. This combines the validation of input and calendar into one API for the consumer, rather having them own this responsibility.